packages/govern-subgraph/src/lib/MiniACL.ts

```diff
@@ -0,0 +1,71 @@
+import { Address, Bytes } from "@graphprotocol/graph-ts";
+import { Role as RoleEntity } from "../../generated/schema";
+
+export function frozenRoles(roles: string[], selector: Bytes): void {
+  roles
+    .map<RoleEntity>((role) => RoleEntity.load(role)!)
+    .filter((role) => role.selector === selector)
+    .forEach((role) => {
+      role.frozen = true;
+      role.save();
+    });
+}
+
+export function roleGranted(
+  entity: Address,
+  selector: Bytes,
+  who: Address
+): RoleEntity {
+  const role = loadOrCreateRole(entity, selector, who);
+  role.granted = true;
+
+  role.save();
+
+  return role;
+}
+
+export function roleRevoked(
+  entity: Address,
+  selector: Bytes,
+  who: Address
+): RoleEntity {
+  const role = loadOrCreateRole(entity, selector, who);
+  role.granted = false;
+
+  role.save();
+
+  return role;
+}
+
+function buildRoleId(entity: Address, selector: Bytes, who: Address): string {
```

Comment:
Having this kind of helpers to help is build the id of an entity I found it super nice. Also a pro tip from Facu 😎

Reply:
Yeah it's soooo clean! Facu tips are 🔥